nixos/modules/services/web-apps/alist.nix

```diff
@@ -0,0 +1,297 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+let
+  cfg = config.services.alist;
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ moraxyc ];
+  };
+
+  options = {
+    services.alist = {
+      enable = lib.mkEnableOption "alist, a file list program";
+      debug = lib.mkEnableOption "debug mode of alist";
+
+      user = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+        description = "Alist user name. If this is not set, a user named `alist` will be created.";
+      };
+
+      group = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+        description = "Alist group name. If this is not set, a group named `alist` will be created.";
+      };
+
+      stateDir = lib.mkOption {
+        type = lib.types.str;
+        default = "/var/lib/alist";
+        description = "Alist stores data and config file in this directory.";
+      };
+
+      allowModify = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Allow Alist to persist settings in the config file.";
+      };
+
+      package = lib.mkPackageOption pkgs "alist" { };
+
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            jwt_secret = lib.mkOption {
+              type = lib.types.attrsOf lib.types.path;
+              example = lib.literalExpression ''
+                {
+                  _secret = "/run/agenix/alist-jwt";
```

Comment:
```suggestion
                  _secret = /run/agenix/alist-jwt;
```

Reply:
@Moraxyc I think if you prefer strings as paths, there's no need to use `lib.literalExpression`, you can simply do `example = {_secret = ...;}`, like in https://github.com/NixOS/nixpkgs/blob/66af7da8a6962c0ccfc42e7271e55bba7ef29a60/nixos/modules/services/web-apps/akkoma.nix#L573